### PR TITLE
Consult refactor

### LINKED ISSUE #1: Why is this a consult-* package?
Hi, I just came across your package and I looked into it! There is no need to use Consult here. Instead you could create a generic package without a dependency on Consult. Instead of `consult--read` use `completing-read`. You could name your package `tramp-completion`.

https://github.com/Ladicle/consult-tramp/blob/main/consult-tramp.el#L118

~~~ elisp
(completing-read "Tramp: " (tramp-completion--candidates))
~~~

### PR DESCRIPTION
**Pull request by elken**
Refactor to use more consult functionality (narrowing, sources, etc) and
provide a slightly more robust featureset.

There is potential to also include some actions for embark, but they
might also be better suited to another package.

Fixes #1

(@minad a review from you here would be welcome :) )

### CODE REVIEW

File: consult-tramp.el [hunk 1]
```diff
@@ -69,54 +79,109 @@
   :group 'consult-tramp
   :type 'string)
 
-(defcustom consult-tramp-sudo-path "/"
-  "Path to connect to the local root."
+(defcustom consult-tramp--local-hosts '("/sudo:root@localhost:/")
```
Comment (minad) [inline consult-tramp.el]: The defcustom should probably not be private consult-tramp-local-hosts?

Comment (elken) [reply consult-tramp.el]: No you're right, it shouldn't be

**Comment by minad** [comment]
It looks to me that the sources from here can be integrated with [consult-dir](https://github.com/karthink/consult-dir). Maybe this should be documented in the README? It is a strong reason to implement this package in this way with Consult sources.

(EDIT: One could also consider adding the sources to consult-dir instead? But this is a matter of preference of how fine-grained these packages should be. Maybe not everyone who uses consult-dir would also want the tramp sources.)

**Comment by elken** [comment]
I thought that when I was working on it, there is a lot of crossover. 

You're right though that not everyone who wants consult-dir would also want the tramp sources, which I think is where this would sit.

On reflection, this might indeed be a better fit as part of consult-dir. I'll leave this up to @Ladicle to decide to merge this or not, but I'll also add the sources to consult-dir I think.

**Comment by minad** [comment]
From a maintenance perspective and reduction of work it is probably better to make this part of consult-dir. This package already exists and the sources just fit exactly there. If something must be updated it is better to have to update it only at a single place.

File: consult-tramp.el [hunk 2]
```diff
@@ -69,54 +79,109 @@
...
+  "Local host candidate source for `consult-tramp'.")
+
+(defvar consult-tramp--source-known-host
+  `(:name     "Known Hosts"
+    :narrow   ?k
+    :category tramp
+    :face     consult-tramp-host
+    :history  consult-tramp-history
+    :items    ,#'consult-tramp--known-hosts)
+  "Known Host candiadate source for `consult-tramp'.")
```
Comment (minad) [inline consult-tramp.el:144]: Typo candiadate

**Comment by elken** [comment]
> From a maintenance perspective and reduction of work it is probably better to make this part of consult-dir. This package already exists and the sources just fit exactly there. If something must be updated it is better to have to update it only at a single place.

Yeah I think you're right again :p

File: consult-tramp.el [hunk 3]
```diff
@@ -44,23 +64,13 @@
   :group 'consult-tramp
   :type 'string)
 
-(defcustom consult-tramp-enable-shosts t
-  "Use known_hosts as a completion source."
-  :group 'consult-tramp
-  :type 'string)
-
 (defcustom consult-tramp-known-hosts "~/.ssh/known_hosts"
```
Comment (minad) [inline consult-tramp.el:67]: I wonder about the known_hosts file. On my system this file is not readable.

Comment (elken) [reply consult-tramp.el:67]: It also doesn't seem to parse anything correctly for me, I'll probably end up leaving it out of the consult-dir PR

image

Comment (minad) [reply consult-tramp.el:67]: Maybe this code was taken from counsel-tramp/helm-tramp which is pretty dated? It is a long time ago that the format of known_hosts changed to the unreadable format.

Comment (elken) [reply consult-tramp.el:67]: Likely, the comment header indicated it was copied and tweaked

**Pull request closed without merging**